From 138dc836ec4c7594361f2864d5bb5510fb94cfe7 Mon Sep 17 00:00:00 2001
From: Tony LIU <junjie.liu@freescale.com>
Date: Fri, 20 Jan 2012 11:02:18 +0800
Subject: [PATCH 0892/2508] ENGR00172475 [USB]Add wakeup entry for USB device
 on Kernel3.0

- in Kernel 3.0, all the wakeup entry is removed by default,
  we need add it manually

Signed-off-by: Tony LIU <junjie.liu@freescale.com>
---
 arch/arm/plat-mxc/usb_common.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/plat-mxc/usb_common.c b/arch/arm/plat-mxc/usb_common.c
index 80e2a95..b50963b 100755
--- a/arch/arm/plat-mxc/usb_common.c
+++ b/arch/arm/plat-mxc/usb_common.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2011 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright (C) 2011-2012 Freescale Semiconductor, Inc. All Rights Reserved.
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -456,7 +456,7 @@ static int usb_register_remote_wakeup(struct platform_device *pdev)
 		return -ENODEV;
 	}
 	irq = res->start;
-	pdev->dev.power.can_wakeup = 1;
+	device_set_wakeup_capable(&pdev->dev, true);
 	enable_irq_wake(irq);
 
 	return 0;
-- 
1.7.7.4

